### PR TITLE
Move linting to eslint/prettier

### PR DESCRIPTION
**Pull request by psoleckimoj**

### CODE REVIEW

File: app/process-tasks.js [hunk 1]
```diff
@@ -6,45 +6,48 @@ const completeTaskWithStatus = require('./services/data/complete-task-with-statu
 const getWorkerForTask = require('./services/get-worker-for-task')
 
 module.exports = function () {
-  const batchSize = parseInt(config.ASYNC_WORKER_BATCH_SIZE)
+  const batchSize = parseInt(config.ASYNC_WORKER_BATCH_SIZE, 10)
```
Comment (psoleckimoj) [inline app/process-tasks.js:9]: `parseInt()` actually requires a radix so you'll see a lot of these.

File: app/process-tasks.js [hunk 2]
```diff
@@ -6,45 +6,48 @@ const completeTaskWithStatus = require('./services/data/complete-task-with-statu
...
+    if (tasks.length === 0) {
+      return Promise.resolve()
+    }
 
-      const promiseArray = []
+    const promiseArray = []
 
-      for (const task of tasks) {
-        const worker = getWorkerForTask(task.task)
+    tasks.forEach(task => {
```
Comment (psoleckimoj) [inline app/process-tasks.js:27]: A few of these converted to `forEach`

File: app/services/auto-approval/auto-approval-process.js [hunk 3]
```diff
@@ -14,104 +14,127 @@ const dateFormatter = require('../date-formatter')
...
+            return result
           })
-      } else {
-        return Promise.resolve(null)
-      }
-    })
+        })
+      })
+    }
+    return Promise.resolve(null)
```
Comment (psoleckimoj) [inline app/services/auto-approval/auto-approval-process.js:99]: A few conversions from a pointless `else`

File: app/services/auto-approval/auto-approval-process.js [hunk 4]
```diff
@@ -14,104 +14,127 @@ const dateFormatter = require('../date-formatter')
...
+  if (
+    autoApprovalData.Claim.ClaimType === claimTypeEnum.FIRST_TIME ||
     autoApprovalData.Claim.ClaimType === claimTypeEnum.REPEAT_NEW_ELIGIBILITY ||
-    (autoApprovalData.Claim.Status !== statusEnum.NEW)) {
+    autoApprovalData.Claim.Status !== statusEnum.NEW
+  ) {
     return true
   }
+
+  return false
```
Comment (psoleckimoj) [inline app/services/auto-approval/auto-approval-process.js:112]: A few missing `returns` added like this

File: app/services/auto-approval/checks/are-children-under-18.js [hunk 5]
```diff
@@ -10,7 +10,7 @@ module.exports = function (autoApprovalData) {
   const now = dateFormatter.now()
 
   if (autoApprovalData.ClaimChildren && autoApprovalData.ClaimChildren.length > 0) {
-    for (let i = 0; i < autoApprovalData.ClaimChildren.length; i++) {
+    for (let i = 0; i < autoApprovalData.ClaimChildren.length; i += 1) {
```
Comment (psoleckimoj) [inline app/services/auto-approval/checks/are-children-under-18.js:13]: A few of these `i++` changes

File: app/services/auto-approval/checks/do-expenses-match-first-time-claim.js [hunk 6]
```diff
@@ -2,24 +2,29 @@ const groupExpensesByType = require('../../notify/helpers/group-expenses-by-type
...
-    for (let i = 0; i < Object.keys(groupedCurrentExpenses).length; i++) {
+    for (let i = 0; i < Object.keys(groupedCurrentExpenses).length; i += 1) {
       const index = Object.keys(groupedCurrentExpenses)[i]
       const currentExpenseTypeGroup = groupedCurrentExpenses[index]
 
       // For each expense type, check that the first time claim contains the same expense type
       // with the same or less number of claim expenses
       if (!groupedFirstTimeClaimExpenses[index]) {
-        const ADDITIONAL_INFO = `. Claim ref: ${autoApprovalData.Claim.Reference}, No. of current expense type: ${currentExpenseTypeGroup.length}, No. of first time claim expense: ${groupedFirstTimeClaimExpenses[index]?.length ?? 0}`
+        const ADDITIONAL_INFO = `. Claim ref: ${autoApprovalData.Claim.Reference}, No. of current expense type: ${currentExpenseTypeGroup.length}, No. of first time claim expense: 0`
```
Comment (psoleckimoj) [inline app/services/auto-approval/checks/do-expenses-match-first-time-claim.js:21]: In the `if` it's already checked so there can't be any first time claim expenses.

File: app/services/auto-approval/checks/do-expenses-match-first-time-claim.js [hunk 7]
```diff
@@ -2,24 +2,29 @@ const groupExpensesByType = require('../../notify/helpers/group-expenses-by-type
...
         return new AutoApprovalCheckResult(CHECK_NAME, false, FAILURE_MESSAGE + ADDITIONAL_INFO)
-      } else if (currentExpenseTypeGroup.length > groupedFirstTimeClaimExpenses[index].length) {
-        const ADDITIONAL_INFO = `. Claim ref: ${autoApprovalData.Claim.Reference}, Expense type: ${JSON.stringify(groupedFirstTimeClaimExpenses[index][0]?.ExpenseType)}, No. of current expense type: ${currentExpenseTypeGroup.length}, No. of first time claim expense: ${groupedFirstTimeClaimExpenses[index]?.length ?? 0}`
+      }
+
+      if (currentExpenseTypeGroup.length > groupedFirstTimeClaimExpenses[index].length) {
+        const firstGroupedFirstTimeClaimExpenses = groupedFirstTimeClaimExpenses[index]
```
Comment (psoleckimoj) [inline app/services/auto-approval/checks/do-expenses-match-first-time-claim.js:26]: Pulled this out to make it easier to read.

File: app/services/get-worker-for-task.js [hunk 8]
```diff
@@ -21,27 +21,46 @@ const sendRequestInformationReminderNotification = require('./workers/send-reque
...
+    case tasksEnum.ARCHIVE_OLD_CLAIMS:
+      return archiveOldClaims
+    case tasksEnum.ARCHIVE_CLAIM:
+      return archiveClaim
+    case tasksEnum.DWP_FAILED_NOTIFICATION:
+      return dwpFailedNotification
+    case tasksEnum.SEND_REQUEST_INFORMATION_REMINDER_NOTIFICATION:
+      return sendRequestInformationReminderNotification
+
+    default:
```
Comment (psoleckimoj) [inline app/services/get-worker-for-task.js:63]: Various `defaults` missing for `switch` statements

File: app/services/workers/generate-direct-payments.js [hunk 9]
```diff
@@ -20,64 +20,63 @@ const generateDirectPayments = function () {
...
+          .then(function () {
+            log.info('Setting all claims as payment processed')
+            return updateAllClaimsProcessedPayment(claimIds, paymentData, true)
+          })
+          .then(function () {
+            log.info('Setting all topups as payment processed')
+            return updateAllTopupsProcessedPayment(topUpClaimIds)
+          })
+      }
+      return Promise.resolve()
```
Comment (psoleckimoj) [inline app/services/workers/generate-direct-payments.js:75]: Some functions had no return

File: test/unit/test-process-tasks.js [hunk 10]
```diff
@@ -37,27 +34,30 @@ describe('process-tasks', function () {
...
-      }
+      },
     })
     mockCompleteTaskWithStatus.mockResolvedValue({})
 
     return processTasks().then(function () {
-      expect(mockGetPendingTasksAndMarkInProgress).toHaveBeenCalledWith('ExtSchema', batchSize) //eslint-disable-line
-      expect(mockGetWorkerForTask).toHaveBeenCalledWith('task1') //eslint-disable-line
-      expect(mockGetWorkerForTask).toHaveBeenCalledWith('task2') //eslint-disable-line
+      expect(mockGetPendingTasksAndMarkInProgress).toHaveBeenCalledWith('ExtSchema', batchSize)
```
Comment (psoleckimoj) [inline test/unit/test-process-tasks.js:51]: I'm not sure why a lot of tests had this in.

File: .eslintcache
Comment (tpmcgowan) [inline .eslintcache:1]: Should this be Git-ignored?

File: app/process-tasks.js (hunk 2 above)
Comment (tpmcgowan) [reply app/process-tasks.js:27]: I didn't know the other syntax worked 😂

**Review by tpmcgowan — APPROVED** [review]
LGTM!

File: app/process-tasks.js (hunk 2 above)
Comment (psoleckimoj) [reply app/process-tasks.js:27]: I'd never actually used it before 😁

**Pull request merged**